vm/src/stdlib/math.rs

```diff
@@ -1,817 +1,820 @@
-/* Math builtin module
- *
- *
- */
-
-use crate::{
-    builtins::{try_bigint_to_f64, try_f64_to_bigint, IntoPyFloat, PyFloatRef, PyInt, PyIntRef},
-    function::{ArgIterable, Args, OptionalArg},
-    utils::Either,
-    PyObjectRef, PyResult, PySequence, TypeProtocol, VirtualMachine,
-};
-use num_bigint::BigInt;
-use num_traits::{One, Signed, Zero};
-use puruspe::{erf, erfc, gamma, ln_gamma};
-use rustpython_common::float_ops;
-use std::cmp::Ordering;
-
-// Helper macro:
-macro_rules! make_math_func {
-    ( $fname:ident, $fun:ident ) => {
-        fn $fname(value: IntoPyFloat, vm: &VirtualMachine) -> PyResult<f64> {
-            let value = value.to_f64();
-            let result = value.$fun();
-            result_or_overflow(value, result, vm)
-        }
-    };
-}
-
-macro_rules! make_math_func_bool {
-    ( $fname:ident, $fun:ident ) => {
-        fn $fname(value: IntoPyFloat) -> bool {
-            value.to_f64().$fun()
-        }
-    };
-}
+use crate::{PyObjectRef, VirtualMachine};
 
-#[inline]
-fn result_or_overflow(value: f64, result: f64, vm: &VirtualMachine) -> PyResult<f64> {
-    if !result.is_finite() && value.is_finite() {
-        // CPython doesn't return `inf` when called with finite
-        // values, it raises OverflowError instead.
-        Err(vm.new_overflow_error("math range error".to_owned()))
-    } else {
-        Ok(result)
-    }
-}
+pub(crate) fn make_module(vm: &VirtualMachine) -> PyObjectRef {
+    let module = math::make_module(vm);
+    let ctx = &vm.ctx;
+    extend_module!(vm, module, {
+        // Constants:
+        "pi" => ctx.new_float(std::f64::consts::PI), // 3.14159...
+        "e" => ctx.new_float(std::f64::consts::E), // 2.71..
+        "tau" => ctx.new_float(2.0 * std::f64::consts::PI),
+        "inf" => ctx.new_float(f64::INFINITY),
+        "nan" => ctx.new_float(f64::NAN)
```

Comment:
these can be `#[pyattr] use f64::{FOO as foo};`, I'm pretty sure. Also I know this is old code but there's now a `std::f64::consts::TAU` constant.

Reply:
Ah, that's even better. You can even use multiple rebindings and things work just fine, didn't condense these down to a single though, seemed too much.